src/components/core-player/hooks/media-type.ts

```diff
@@ -0,0 +1,29 @@
+import { MediaType } from '../../../types/media-type';
+import { isAudio, isUrlSupported } from '../../../utils/is-url-supported';
+
+interface UseCorePlayerHookProps {
+	url?: string;
+	initialMediaType?: string;
+}
+
+export function getMediaType({
+	url,
+	initialMediaType,
+}: UseCorePlayerHookProps): MediaType {
+	if (!url) {
+		return 'unknown';
+	}
+	// If mediaType was initialized external => we use it
+	if (initialMediaType) {
+		return initialMediaType.startsWith('audio/') ? 'audio' : 'video';
+	}
```

Comment:
Here tests fails on `unknown` media type.
If user provided `unknown`, but he URL would be a video url, `MediaType` would be video, instead `unklnown`